source/sync/data-access-patterns/flexible-sync-permissions-guide.txt

```diff
@@ -0,0 +1,611 @@
+between the document creator and the collaborators they add to the document.
+
+However, if a user has write access to a document, they may write to any field
+of the document. Consequently, this strategy **allows collaborators to add other
+collaborators**. It would also allow a collaborator to edit the ``owner_id``
+field. Finer control of permissions are not possible without field-level
+permissions, which are not currently available.
+
+User Search
+~~~~~~~~~~~
+
+How exactly to get the other user's ID depends on the details of your app. For
+example, when a user wants to add another user to a document, you might have a
+search box that accepts an email address. If the given email address corresponds
+to another user, the client can add that user's ID to the document's
+``collaborators`` array.
+
+Realm has no built-in way to search users. Generally, the flow for searching
+users is as follows:
+
+- Set up an authentication trigger to create a user document when a user
+  registers. The user document contains information you'll use to look up later,
+  such as the user's email address.
+- Create a function that queries the user data collection for a user.
+- Call the function from the client side when the user wants to find another user.
+
+To create an authentication trigger, follow these steps:
+
+.. include:: /includes/create-auth-trigger.rst
+
+- 
+   - 
+      .. code-block:: javascript
+         
+          exports = function(authEvent) {
+            const user = authEvent.user;
+
+            //Access the mongodb service for custom user data:
+            const collection = context.services.get("custom-user-data").db("Item").collection("User");
+            const newDoc = {
+              _id: user.id,
+              email: user.data.email, // Assumes email/password authentication
+            };
+            
+            // It's OK to do an insert and not a upsert because this function 
+            // only runs when a new user account is created
+            return collection.insertOne(newDoc);
+         };
+
+Next, create a system function called ``findUser``:
+
+.. include:: /includes/create-a-system-function.rst
+
+In the :guilabel:`Function Editor`, paste the following code and save:
+
+.. code-block:: javascript
+
+   exports = async function(email) { 
+     // Access the mongodb service for custom user data:
+     const collection = context.services.get("custom-user-data").db("Item").collection("User");
+     const filter = {
+       email,
+     };
+     // Search for the user by email
+     const result = await collection.findOne(filter);
+     // Return null or the corresponding user id
+     return result && result._id;
+   };
+
+From your client, you can now call this function. The function's only argument
+is an email address string. If the email corresponds to a user, the function
+returns the user's ID. Otherwise, it returns null.
+
+.. _tiered-privileges:
+
+Tiered Privileges
+-----------------
```

Comment:
i'd also consider doing procedure/step directives for all the distinct steps of this